### DISCUSSION TITLE
Can I import other functions into my Vercel serverless API?

### DISCUSSION THREAD

**Discussion by user72356** [vercel/community#1007] [Help] [answered]
I'm using `vercel dev` to test a Vite React project locally (not a Next.js project). 

In the API folder, I created a simple `index.ts` file and another `test.ts` file. The `index.ts` file contains the route handler for the serverless function and `test.ts` is just a simple function called `test()`.

I can't import `test()` into the `index.ts` file to be used within the API route handler. I get a `Cannot find module './test.js'` error. 

When switching to the old-fashioned Node.js export/require system (BTW this requires adding `"moduleResolution": "Classic",` to the `tsconfig.node.json` file), it works.

What gives? How can I use ES6-style imports to import my own non-handler functions?

**Comment by amyegan** [comment]
Hi @user72356! My best guess is that it has something to do with the build process and whatever tools and packages are used in your app. Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)?

**Comment by NWylynko** [comment, marked as answer]
Im having the same issue, changing the moduleResolution doesn't seem to have any effect. The stack i'm using is Node 18, Vite, React, Typescript, Pnpm and vercel ofc

**Reply by NWylynko** [reply]
Changing the imports from `import { function } from "./module"` to `import { function } from "./module.js"` fixes the issue
